src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -0,0 +1,292 @@
+     * @param workerOperations        the workerOperations, used to do operations on workers
+     * @param notificationMailAddress the mail address to send notifications
+     */
+    public MoneyTransferManager(MailHandler mailHandler, WorkerBalanceOperations workerBalanceOperations, WorkerOperations workerOperations, String notificationMailAddress) {
+        this.mailHandler = mailHandler;
+        this.workerOperations = workerOperations;
+        this.workerBalanceOperations = workerBalanceOperations;
+        this.payOffThreshold = 0;
+        this.notificationMailAddress = notificationMailAddress;
+        this.notificationText = new StringBuilder();
+    }
+
+    /**
+     * Starts the MoneyTransferManager, so giftcodes become submitted to workers every 7 days.
+     */
+    public synchronized void start() {
+        if (schedule != null) {
+            throw new IllegalStateException("run() was called twice!");
+        }
+
+        Runnable runnable = () -> {
+            try {
+                submitGiftCodes();
+            } catch (MoneyTransferException e) {
+                try {
+                    sendNotification(e.toString(), true);
+                } catch (MoneyTransferException f) {
+                    LOGGER.error(f);
+                }
+                LOGGER.error(e);
+            }
+        };
+
+
+        schedule = scheduler.scheduleAtFixedRate(runnable, 7, 7, TimeUnit.DAYS);
+
+    }
+
+    /**
+     * Shuts the MoneyTransferManager down.
+     */
+    public synchronized void shutdown() {
+        schedule.cancel(false);
+        scheduler.shutdown();
+        schedule = null;
+    }
+
+    /**
+     * Logs a new money transfer and saves it.
+     *
+     * @param workerID the id of the worker, who gets the money
+     * @param amount   the amount of money in ct
+     */
+    public void addMoneyTransfer(int workerID, int amount, int expID) {
+        workerBalanceOperations.addCredit(workerID, amount, expID);
+    }
+
+    /**
+     * Pays all workers depending on their logged money transfers.
+     * @throws MoneyTransferException gets thrown, if an error occurred
+     */
+    public void submitGiftCodes() throws MoneyTransferException {
+        LOGGER.trace("Started fetching new giftcodes.");
+        fetchNewGiftCodes();
+        LOGGER.trace("Completed fetching new giftcodes.");
+
+        LOGGER.trace("Started submission of giftcodes to workers.");
+        Result<WorkerRecord> workers = workerOperations.getWorkerWithCreditBalanceGreaterOrEqual(payOffThreshold);
+        List<GiftCodeRecord> giftCodes = workerBalanceOperations.getUnusedGiftCodes();
+
+        for (WorkerRecord worker : workers) {
+            if (!worker.getEmail().equals("")) {
+                List<GiftCodeRecord> payedCodesForWorker = chooseGiftCodes(worker, giftCodes);
+
+                giftCodes = workerBalanceOperations.getUnusedGiftCodes();
```

Comment:
shouldnt this happen _after_ payWorker ? i cannot see where in chooseGiftCodes the codes are getting marked used ...